src/customer/customer-strategy-reducer.ts

```diff
@@ -174,6 +187,19 @@ function statusesReducer(
             signOutMethodId: undefined,
         });
 
+    case CustomerStrategyActionType.ExecutePaymentMethodCheckoutRequested:
+        return objectMerge(statuses, {
+            isExecutingPaymentMethodCheckout: true,
+            executePaymentMethodCheckoutMethodId: action.meta && action.meta.methodId,
+        });
+
+    case CustomerStrategyActionType.ExecutePaymentMethodCheckoutFailed:
+    case CustomerStrategyActionType.ExecutePaymentMethodCheckoutSucceeded:
```

Comment:
Are there any errors that we need to catch and set in case of faliure?

Reply:
We catch all the errors in `_executePaymentMethodProvider` method in `BoltCustomerStrategy` and throw them to checkout page, where we handle them in those way that we need. 

I think that all of the errors in the '_executePaymentMethodProvider' method will be about failures on provider service side or about provided arguments errors from checkout side.